packages/apollo-collaboration-server/src/features/features.service.ts

```diff
@@ -196,19 +212,38 @@ export class FeaturesService {
 
     const { assembly } = exportDoc
     const refSeqs = await this.refSeqModel.find({ assembly }).exec()
+    const refSeqIds = refSeqs.map((refSeq) => refSeq._id)
+    let printFasta = true
 
     const headerStream = new Readable({ objectMode: true })
+    const sequenceStream = new Readable({ objectMode: true })
+
+    // Implement the _read() method for the custom stream. This is just a placeholder; we can leave it empty
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    headerStream._read = function () {}
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    sequenceStream._read = function () {}
+
     headerStream.push('##gff-version 3\n')
     for (const refSeqDoc of refSeqs) {
       headerStream.push(
         `##sequence-region ${refSeqDoc.name} 1 ${refSeqDoc.length}\n`,
       )
+      for await (const doc of this.refSeqChunksModel
+        .find({ refSeq: refSeqDoc.id })
+        .sort({ n: 1 })
+        .cursor()) {
```

Comment:
Each ref seq might contain multiple ref seq chunks, so you'll need to take that into account. For example, if I set `CHUNK_SIZE` in the env file to 20,000 and then import a Volvox assembly, the exported FASTA looks like this:
```
##FASTA
>ctgA 
cattgttgcggagttgaacaACGGCATTAGGAACACTTCCGTCTCtcacttttatacgat
...
gtaagcgataggcaaactgcggttattcgcgtagaccgcttgatgatgtaaaataacagt
##FASTA
>ctgA 
cattgttgcggagttgaacaACGGCATTAGGAACACTTCCGTCTCtcacttttatacgat
...
cgaacaccgtgatccattga
>ctgA
ggcgtggaggcatagggccggagtgttcccgtaccaattacgagctaagaccgcctgtcc
...
aaaaacgaattacgtgtatagagattatgtcactccgtcac
>ctgB
ACATGCTAGCTACGTGCATGCTCGACATGCATCATCAGCCTGATGCTGATACATGCTAGC
...
```
So there are multiple ctgAs, and ##FASTA shows up twice somehow.

Reply:
Now, there is just one time **##FASTA** and RefSeqs name and description even sequence was saved in several chunks in Mongo. However, due to multiple chunks per sequence, sequence lines are not always same length i.e. output may look like this:
_...
tgggtagagcgccgatagctaattctcgagtggaactgcggaatgtgtcatatgaggtag
gcacggccctggcgtggcggcggtacgcgtacaaacctccactgcttggggaataaggtg
cgaacaccgtgatccattga
ggcgtggaggcatagggccggagtgttcccgtaccaattacgagctaagaccgcctgtcc
agccctggtgattacgtagtaagctcggtacaagcctgtaattagcttccgggtggcgga
cattcgttcatattgccagtcacggcagaggccgcgctagctaggccagccccgatgaaa
..._

I did not find any spec that each sequence line should be same length so I guess this is not an issue?

Reply:
They do need to be the same length (except for the last line), not necessarily because of the spec, but because most parsers/indexers that people use on this output will expect them the be the same length.

Reply:
Now added the following property
# Sequence line lenght in exported GFF3 file
SEQ_LINE_LEN=90

All lines (except the last one) are now this long